### ISSUE TITLE
Kubernetes v1.18.0 cpumanager: cpuset not assigned after pod restart 

### ISSUE THREAD

**Issue by joshshn** [kubernetes/kubernetes#91668]
**What happened**:
Deployed pod with guarenteed QoS class in a kubernetes v1.18.0 cluster having Static CPU Management policy configured. Pod runs fine with effective cpuset matching the cpu resource in pod manifest. However when pod restarts( in this case due to liveness probe failure) the cpumanager does not assign cpuset for the pod.

**What you expected to happen**:
The cpumanager should allocate the requested number of cpus for the pod even after pod restart.

**How to reproduce it (as minimally and precisely as possible)**:

Run sample pod with manifest:

```
apiVersion: v1
kind: Pod
metadata:
  labels:
    test: liveness
  name: liveness-exec
spec:
  containers:
  - name: liveness
    image: k8s.gcr.io/busybox
    args:
    - /bin/sh
    - -c
    - touch /tmp/healthy; sleep 120; rm -rf /tmp/healthy; sleep 6000
    resources:
      limits:
        cpu: "5"
        memory: 1Gi
      requests:
        cpu: "5"
        memory: 1Gi
    livenessProbe:
      exec:
        command:
        - cat
        - /tmp/healthy
      initialDelaySeconds: 5
      periodSeconds: 5
```

Login to worker node where pod is deployed and check the cpumanager state:

```
ubuntu@worker-1:~$ sudo cat /var/lib/kubelet/cpu_manager_state
{"policyName":"static","defaultCpuSet":"0-3,11-27,29-31,39-55","entries":{"03b18c6b-cc02-48e7-9d04-a856f64ad624":{"cids":"4-8,32-36"},"0ae5ab33-6279-4f96-b287-1d54b8ce76c9":{"liveness":"9-10,28,37-38"}},"checksum":4207955599}
```
Here it is seen that the container is assigned the cpuset - "liveness":"9-10,28,37-38"
Same is verifed in the pod:
```
ubuntu@master-1:~$ kubectl exec -it liveness-exec -- cat /sys/fs/cgroup/cpuset/cpuset.effective_cpus
9-10,28,37-38
```

Wait for pod to restart after 120 seconds due to liveness probe failure and recheck the cpumanager state:
```
ubuntu@worker-1:~$ sudo cat /var/lib/kubelet/cpu_manager_state
{"policyName":"static","defaultCpuSet":"0-3,9-31,37-55","entries":{"03b18c6b-cc02-48e7-9d04-a856f64ad624":{"cids":"4-8,32-36"}},"checksum":3637020572}
```
Same is verified from the pod:
ubuntu@master-1:~$ kubectl exec -it liveness-exec -- cat /sys/fs/cgroup/cpuset/cpuset.effective_cpus
0-3,9-31,37-55

**Environment**:
- Kubernetes version (use `kubectl version`): v1.18.0
- Cloud provider or hardware configuration: Baremetal Hardware
- OS (e.g: `cat /etc/os-release`):Ubuntu 18.04.3 LTS
- Kernel (e.g. `uname -a`): 5.0.0-37-generic

**Comment by Yukariko** [comment]
@joshshn , see #90377
That pr are merged to v1.19.0.beta.
I did cherry-pick that commit into my branch release-1.18.2, then problem is solved.

**Comment by harper1011** [comment]
@Yukariko  thanks for sharing this PR.
BTW: is there any workaround available for this issue other than cherrypick or upgrade to the release which contains the correction?

**Comment by Yukariko** [comment]
@harper1011 , I don't know another solution.
But I saw other issues, It is fixed in kubelet 1.18.4 
see  https://github.com/kubernetes/kubernetes/issues/92436#issuecomment-653079508
